client/chat/channel-menu-items.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useEffect } from 'react'
+import styled from 'styled-components'
+import { ChannelModerationAction } from '../../common/chat'
+import { appendToMultimap } from '../../common/data-structures/maps'
+import { MULTI_CHANNEL } from '../../common/flags'
+import { SbUserId } from '../../common/users/sb-user'
+import { openDialog } from '../dialogs/action-creators'
+import { DialogType } from '../dialogs/dialog-type'
+import { MenuItem } from '../material/menu/item'
+import { useAppDispatch, useAppSelector } from '../redux-hooks'
+import { openSnackbar } from '../snackbars/action-creators'
+import { useStableCallback } from '../state-hooks'
+import { colorError } from '../styles/colors'
+import { MenuItemCategory } from '../users/user-context-menu'
+import { getChatUserProfile, moderateUser } from './action-creators'
+
+const DestructiveMenuItem = styled(MenuItem)`
+  color: ${colorError};
+
+  --sb-ripple-color: ${colorError};
+`
+
+/**
+ * Even though this function is technically not a React component, nor a custom hook, we still treat
```

Comment:
nit: This isn't really appropriate for being a doc comment. Doc comments are mainly for people *using* the function, this can just be a note above it or whatever

Reply:
We could also honestly consider this to be a hook and just rename the parameter to like `useModifiedMenuItems`, which would make the linter happy as well (provided you also renamed this to match). May be easier than trying to fight it

Reply:
The reason why I didn't name it as a hook is that we're calling it inside another hook, which not only triggers another linter error, but makes it pretty confusing to read as well :d

Reply:
Calling it inside a hook should be fine, tons of hooks are just forwarding to other hooks (e.g. `useState`, `useRef`, etc.), so I think you may be confused about the real reason for the lint error. I don't really care very much whether it is named that way or we just disable the lint though, so whatever